### DISCUSSION TITLE
How to get the name of a playlistitem?

### DISCUSSION THREAD

**Discussion by YihaoChen96** [demberto/PyFLP#83] [Q&A] [answered]
Dear Dev team,
Thank you all for the hard work!
Recently I encountered a problem when using PyFLP. I was iterating through the Arrangement and got a set of Playlistitems. I wanted to retrieve the names of the items and match them with patterns but didn't find a way to do so. 
May I ask if there's any way I can get the names of the items acquired through Arrangement? Really appreciate it!

**Comment by demberto** [comment, marked as answer]
Yea, I planned to make use of `ChannelPlaylistItem` and `PatternPlaylistItem` classes, but until that's done you can use this code:

```py
import pyflp

project = pyflp.parse(r"F:\Prog\Python\PyFLP\tests\assets\FL 20.8.4.flp")

for track in (track for arr in project.arrangements for track in arr.tracks):
    # print(track) -- Get track information
    for playlist_item in track:
        if playlist_item["item_index"] <= playlist_item["pattern_base"]:
            channel_iid = playlist_item["item_index"]
            print(f"Playlist item for {project.channels[channel_iid]}")
        else:
            pattern_num = (
                playlist_item["item_index"] - playlist_item["pattern_base"]
            )  # 1, 2, 3, ...
            print(f"Playlist item for {project.patterns[pattern_num]}")
```

I just opened an issue (#84). You can subscribe for notifications related to that issue.
If you get any errors or have any questions, please feel free to share them :)

**Reply by YihaoChen96** [reply]
Thank you so much for your swift reply Demberto! I have already subscribed and looking forward to it!

**Reply by demberto** [reply]
@YihaoChen96 Check the latest commits, feature implemented
